### PR TITLE
ContainerApps 04/19 Release

### PR DESCRIPTION
**Pull request by calvinsID**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
ContainerApps

**Comment by StrawnSC** [comment]
@calvinsID can you un-draft this when you get the chance -- the GH actions update is good to go

The GitHub action didn't run successfully when I used a non-ACR registry (though ACR works great). I ran into the same issue when I tried with the previous API version so there wasn't a regression in the API

**Comment by panchagnula** [comment]
@zhoxing-ms can we get an initial review from you please, so we can get this ready for 4/19 release after further testing on our end.

**Comment by panchagnula** [comment]
@StrawnSC can you update the .github/CODEOWNERS to include your alias to the owners list as well please. Thanks!

File: src/containerapp/azext_containerapp/_clients.py
Comment (zhoxing-ms) [inline src/containerapp/azext_containerapp/_clients.py:530]: Why hard code the version here instead of upgrading the version of `NEW_API_VERSION`?  We usually use one version according to the RP dimension, because it is easy to maintain

File: src/containerapp/azext_containerapp/_clients.py
Comment (zhoxing-ms) [inline src/containerapp/azext_containerapp/_clients.py:559]: Same as above

File: src/containerapp/azext_containerapp/_clients.py
Comment (panchagnula) [reply src/containerapp/azext_containerapp/_clients.py:530]: @zhoxing-ms We will have 2 versions for a while until we test all commands with stable api-version '2022-03-01'
The NEW_API_VERSION points to the preview version & we will move out of this in time for GA. Once we have all tests covered we will make the change as a separate PR. A lot of the nee features we are adding support for will be with the non-preview version, like this updates to GitHub actions. Hope that explains.

File: src/containerapp/azext_containerapp/_clients.py
Comment (panchagnula) [reply src/containerapp/azext_containerapp/_clients.py:559]: @StrawnSC we should probably name these preview-version & stable-version, so the swap when we need to do is easy. "NEW_API_VERSION"  name here is misleading, since this is the not the new version.

Comment (StrawnSC) [reply src/containerapp/azext_containerapp/_clients.py]: Just resolved this @panchagnula @zhoxing-ms

File: src/containerapp/HISTORY.rst
Comment (panchagnula) [inline src/containerapp/HISTORY.rst:8]: lets explicitly call out docker-path replaced by context-path

File: src/containerapp/azext_containerapp/tests/latest/test_containerapp_scenario.py
Comment (panchagnula) [inline src/containerapp/azext_containerapp/tests/latest/test_containerapp_scenario.py:21]: not all subs have permissions to EUAP regions, & these tests don't run on our subs so avoid using this region. Use our non-canary region for testing.

**Review by panchagnula — APPROVED** [review]
One comment - otherwise LGTM

File: src/containerapp/azext_containerapp/tests/latest/test_containerapp_scenario.py
Comment (StrawnSC) [reply src/containerapp/azext_containerapp/tests/latest/test_containerapp_scenario.py]: Changed this to "eastus2"

File: src/containerapp/HISTORY.rst
Comment (StrawnSC) [reply src/containerapp/HISTORY.rst]: Just fixed this

**Comment by StrawnSC** [comment]
> @calvinsID can you un-draft this when you get the chance -- the GH actions update is good to go
> 
> The GitHub action didn't run successfully when I used a non-ACR registry (though ACR works great). I ran into the same issue when I tried with the previous API version so there wasn't a regression in the API

@panchagnula update on this: I was erroneously using `--registry-url docker.io/<username>` instead of `--registry-url docker.io`. It works correctly now. That's my bad; the API is functional here.

File: src/containerapp/azext_containerapp/_params.py
Comment (zhoxing-ms) [inline src/containerapp/azext_containerapp/_params.py:138]: The default value of `type` is `str`, so there is no need to define it repeatedly
```suggestion
        c.argument('image', options_list=['--image', '-i'], help="Container image name that the Github Action should use. Defaults to the Container App name.")
```

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**
